### ISSUE TITLE
DNS search suffix list is incomplete on Windows

### ISSUE THREAD

**Issue by PatrickLang** [Azure/azure-container-networking#206]
**Is this an ISSUE or FEATURE REQUEST?** (choose one): Issue

---

**Which release version?**: 1.0.7

---

**Which component (CNI/IPAM/CNM/CNS)**: CNI

---

**Which Operating System (Linux/Windows)**: Windows Server version 1803 (17134)

---

**Which Orchestrator and version (e.g. Kubernetes, Docker)**: Kubernetes v1.11.1

---

**What happened**:

Windows pods still only get the dns search suffix for their namespace. They should also have `svc.cluster.local` and `cluster.local` so that services can be resolved for other namespaces.

Here's `/etc/resolv.conf` on Linux

```
kubectl exec -i busybox-sleep cat /etc/resolv.conf
nameserver 10.0.0.10
search default.svc.cluster.local svc.cluster.local cluster.local g5hqkyfpy0nuzckf3igv405gkg.xx.internal.cloudapp.net
options ndots:5
```

Windows should have the same DNS search suffix results, but it only has the current namespace:

```
$ kubectl exec iis-1803-688b96c57b-s6657 ipconfig

Windows IP Configuration


Ethernet adapter vEthernet (f6667ee9-eth0):

   Connection-specific DNS Suffix  . : default.svc.cluster.local
   Link-local IPv6 Address . . . . . : fe80::8193:c234:bebb:93db%27
   IPv4 Address. . . . . . . . . . . : 10.240.0.136
   Subnet Mask . . . . . . . . . . . : 255.240.0.0
   Default Gateway . . . . . . . . . : 10.240.0.1
```

---

**How to reproduce it** (as minimally and precisely as possible):

1. Deploy a cluster with acs-engine 0.19.3 or newer. 0.20.0 preferred
2. `kubectl apply -f https://gist.githubusercontent.com/PatrickLang/0df013d20d32eb98bc57456c4f73461a/raw/052470dc51c9d3582e85e370d2fb1adca78345af/iis-1803-healthcheck.yaml`
3. `kubectl get pod` - find the pod once its running
4. `kubectl exec podname ipconfig`

---

**Anything else we need to know**:

This is somewhat related to #146. That PR added the pod's namespace, but not the others. If you need more info on what needs to change, please check with @dineshgovindasamy  and @daschott

---

**Comment by yuedai** [comment]
@PatrickLang, can you explain more about your last statement? Do you mean we can expect a fix from new Kubernetes release?

**Comment by PatrickLang** [comment]
Yes, will fix in Kubernetes v1.12 and backport to v1.11 and maybe v1.10.
